### PR TITLE
Feat/convert mdbook/144

### PR DESCRIPTION
**Pull request by mehtaishita**
Problem
=======
Our static site generator (docz) isn't really meeting our needs currently. 
#144

Solution
========
Using mdBook as our static site generator to take advantage of improved functionality: 
- Nested menus in the sidebar
- Multiple themes
- Better configuration opportunities
- Multiple third-party plugins

with @wilwade 

Change summary:
---------------
This is Pull request 1 of 3. See comment for next steps
- [x] Updated dependencies in package.json
- [x] Removed docz and generated spec with mdBook
- [x] Configured styles to have a cleaner look
- [x] Add custom var for sidebar menu folding and collapsing nested menus
- [x] Added third-party plugin [mdBook-linkcheck](https://github.com/Michael-F-Bryan/mdbook-linkcheck) for checking broken links 
- [x] Added github repo URL to the menu bar for more transparency 
- [x] Replaced absolute URLs with relative ones 
- [x] Used provided handlebars to add DSNP logo, and external link to the developer portal 

Steps to Verify:
----------------
From root dir:
1. npm install clean
1. npm run build
2. Build process will run a link-check to check for broken links
1. npm run lint
1. npm run serve
1. The spec will automatically open in your web browser
1. Click around for internal and external links

Screenshots (optional):
-----------------------
image

### CODE REVIEW

**Comment by mehtaishita** [comment]
- [ ] Styling issues - ensure colors and highlights are more consistent
- [ ] The default theme when building/serving is not consistent
- [ ] The github edit link for index.md is broken - however the rest are working fine
- [ ] The developer portal link is not formatted consistently as the others
- [ ] Explore available third party plugins to also show headers in the sidebar toc

File: pages/ActivityContent/Associated/Attachments.md [hunk 1]
```diff
@@ -146,11 +140,11 @@ Audio objects must include a `"hash"` field as described in the [content proofs
...
 | `mediaType` | [Activity Vocabulary 2.0](https://www.w3.org/TR/activitystreams-vocabulary/#dfn-mediatype) | YES | MIME type of `href` content | |
-| `hash` | [DSNP 1.0](Hash) | YES | Array of hashes for linked content validation | MUST include at least one [supported hash](Hash#supported-algorithms)  |
+| `hash` | [DSNP 1.0](Hash.md) | YES | Array of hashes for linked content validation | MUST include at least one [supported hash](Hash.md#supported-algorithms)  |
 | `height` | [Activity Vocabulary 2.0](https://www.w3.org/TR/activitystreams-vocabulary/#dfn-height) | no | A hint as to the rendering height in device-independent pixels |  |
-| `width` | [Activity Vocabulary 2.0](https://www.w3.org/TR/activitystreams-vocabulary/#dfn-height) | no | A hint as to the rendering width in device-independent pixels |  |
+| `width` | [Activity Vocabulary 2.0](https://www.w3.org/TR/activitystreams-vocabulary/#dfn-width) | no | A hint as to the rendering width in device-independent pixels |  |
```
Comment (wilwade) [inline pages/ActivityContent/Associated/Attachments.md:147]: 🥇

File: theme/index.hbs [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+                sidebar = sidebar || 'visible';
+            }
+            html.classList.remove('sidebar-visible');
+            html.classList.add("sidebar-" + sidebar);
+        </script>
+
+        <nav id="sidebar" class="sidebar" aria-label="Table of contents">
+            <div class="sidebar-scrollbox">
+                <a href="https://www.dsnp.org/">
+                    <img src="{{ path_to_root }}DSNP_Logo.png" alt="The DSNP Logo" width="123" height="55"/>
```
Comment (wilwade) [inline theme/index.hbs:107]: I see two DSNP_Logo.png files in the PR. One in `images` and one in `pages`. I would expect the `images` one to be the correct one, but I am not sure.

File: README.md [hunk 3]
```diff
@@ -30,13 +30,19 @@ To run the spec generator locally and preview the formatted spec website, use th
 
 ```
```
Comment (wilwade) [inline README.md:31]: You have to install mdBook as well right? We should mention that and the specific version we expect you to use and link to the how to install it.

Comment (wilwade) [reply README.md:31]: We should also have the prereq of https://github.com/Michael-F-Bryan/mdbook-linkcheck

I wonder if there is a way to make it such that just a `cargo install` can install both mdbook and mdbook-linkcheck. Then we just have the rust requirement and `cargo install` and `npm install`.

File: theme/favicon.svg [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 199.7 184.2">
```
Comment (wilwade) [inline theme/favicon.svg:1]: We should remove these default favicons (png and svg). We have a favicon.ico under `public` that we should use.

File: pages/summary.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
+# Summary
+
+[Introduction](Index.md)
```
Comment (wilwade) [inline pages/summary.md:3]: macOS is not case sensitive (by default), but our server likely is. We should either rename this `index.md` or rename the file `index.md -> Index.md`

File: pages/summary.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+- [Activity Content](ActivityContent/Overview.md)
+    - [Note](ActivityContent/Types/Note.md)
+    - [Profile](ActivityContent/Types/Profile.md)
+    - [Associated Type]()
+        - [Hash](ActivityContent/Associated/Hash.md)
+        - [Location](ActivityContent/Associated/Location.md)
+        - [Tag](ActivityContent/Associated/Tag.md)
+        - [Attachments](ActivityContent/Associated/Attachments.md)
+- [Glossary](Reference/Glossary.md)
+- [Draft Specifications](Draft/Archivists.md)
```
Comment (wilwade) [inline pages/summary.md:36]:
```suggestion
- [Draft Specifications]()
  - [Archivists](Draft/Archivists.md)
```

**Review by wilwade — CHANGES_REQUESTED** [review]
Looking really good. I'm really glad we are not trying to smash in all the css changes into this PR.

Few small changes should get the ci check working as well. (With just the `npm lint` stuff for now)

File: README.md (hunk 3 above)
Comment (saraswatpuneet) [reply README.md:31]: Yes this is a good place to setup linting for rust projects we can use the same outcome in mrc repo

File: book.toml [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+git-repository-url = "https://github.com/LibertyDSNP/spec"
+edit-url-template = "https://github.com/LibertyDSNP/spec/blob/main/{path}"
+
+# https://github.com/Michael-F-Bryan/mdbook-linkcheck
+[output.linkcheck]
+follow-web-links = true 
+traverse-parent-directories = false
+
+[output.html.fold]
+enable = true
```
Comment (saraswatpuneet) [inline book.toml:20]: these extra line lints/pretty

File: package.json [hunk 8]
```diff
@@ -6,37 +6,25 @@
...
-        "build": "docz build",
-        "serve": "docz serve",
+        "build": "mdbook build -o",
+        "serve": "mdbook serve -o",
+        "watch": "mdbook watch -o",
         "spellcheck": "mdspell './pages/**/*.md' './*.md' '!./LICENSE.md' --en-us --ignore-numbers --report",
         "spellfix": "mdspell './pages/**/*.md' './*.md' '!./LICENSE.md' --en-us --ignore-numbers",
         "lint-md": "remark --frail ./pages/**/*.md ./pages/*.md",
         "lint": "npm run lint-md && npm run spellcheck"
     },
```
Comment (saraswatpuneet) [inline package.json:20]: are we going to retire npm/docz @wilwade  @mehtaishita

Comment (saraswatpuneet) [reply package.json:20]: mostly linting is used from npm perspective so it can be removed with how we can lint with cargo and get rid of docz completely

**Review by saraswatpuneet — APPROVED** [review]
Looks good just need to fix linting issues and rest of the work is tracked with other stories nevertheless
Good work :1st_place_medal:

File: theme/index.hbs (hunk 2 above)
Comment (mehtaishita) [reply theme/index.hbs:107]: Yes the images one is correct. Once I upload the logo to the images dir, I can use the public github link as the src

File: README.md (hunk 3 above)
Comment (mehtaishita) [reply README.md:31]: I'll add the commands to readme for now. I believe we can set up cargo config to list the packages

File: theme/index.hbs (hunk 2 above)
Comment (wilwade) [reply theme/index.hbs:107]: Let's put the pages logo into `pages/images/...` and just reference it from there.

File: pages/summary.md (hunk 5 above)
Comment (saraswatpuneet) [reply pages/summary.md:3]: @mehtaishita  linux is case sensitive so SUMMARY.md -> summary.md in linux wont compile and i believe so for windows
Fix : let it be SUMMARY.md

**Review by saraswatpuneet — CHANGES_REQUESTED** [review]
@mehtaishita  Linux is case sensitive so SUMMARY.md -> summary.md in linux wont compile and most our images in build will be run on Debian
Fix : let it be SUMMARY.md

**Review by wilwade — APPROVED** [review]
🚢

**Review by saraswatpuneet — APPROVED** [review]

**Pull request merged**